code/scripting/api/libs/base.cpp

```diff
@@ -26,38 +26,60 @@
 #include "utils/Random.h"
 #include "cmdline/cmdline.h"
 
+
 namespace scripting {
+extern SCP_string ade_tostring(lua_State* L, int argnum, bool add_typeinfo);
+	
 namespace api {
 using Random = ::util::Random;
 
 //**********LIBRARY: Base
 ADE_LIB(l_Base, "Base", "ba", "Base FreeSpace 2 functions");
 
-ADE_FUNC(print, l_Base, "string Message", "Prints a string", NULL, NULL)
+ADE_FUNC(print, l_Base, "string Message", "Prints a string", nullptr, nullptr)
+{
+#ifndef NDEBUG
+	auto str = ade_tostring(L, -1, false);
+	nprintf(("scripting", "%s", str.c_str()));
+#else
+	SCP_UNUSED(L);
+#endif
+
+	return ADE_RETURN_NIL;
+}
+
+ADE_FUNC(println, l_Base, "string Message", "Prints a string with a newline", nullptr, nullptr)
 {
-	nprintf(("scripting","%s", lua_tostring(L, -1)));
+#ifndef NDEBUG
+	auto str = ade_tostring(L, -1, false);
+	nprintf(("scripting", "%s\n", str.c_str()));
+#else
+	SCP_UNUSED(L);
+#endif
 
 	return ADE_RETURN_NIL;
 }
 
-ADE_FUNC(warning, l_Base, "string Message", "Displays a FreeSpace warning (debug build-only) message with the string provided", NULL, NULL)
+ADE_FUNC(warning, l_Base, "string Message", "Displays a FreeSpace warning (debug build-only) message with the string provided", nullptr, nullptr)
 {
+	auto str = ade_tostring(L, -1, false);
+
 	if (Cmdline_lua_devmode) {
-		nprintf(("scripting","WARNING: %s\n", lua_tostring(L, -1)));
-	}
-	else {
-		Warning(LOCATION, "%s", lua_tostring(L, -1));
+		nprintf(("scripting", "WARNING: %s\n", str.c_str()));
+	} else {
+		Warning(LOCATION, "%s", str.c_str());
 	}
```

Comment:
If you ifndef NDEBUG the print / println calls, why not ifndef the warning as well, since it'll only do stuff in debug mode anyways? Could save some performance, especially with heavy use of warning calls for debug